docs/content/advanced/backend/events/architecture.md

```diff
@@ -0,0 +1,223 @@
+
+### subscribe
+
+The `subscribe` method will attach a handler method to the specified event, which is run when the event is triggered. It is usually used insde a subscriber class.
+
+The `subscribe` method accepts the event name as the first parameter. This is the event that the handler method will attach to.
+
+The second parameter is the handler method that will be triggered when the event is emitted.
+
+The third parameter is an optional `context` parameter. It allows you to configure the ID of the handler method.
+
+The `subscribe` method has the following signature:
+
+```ts
+export default class EventBusService {
+ // ...
+ subscribe(
+    event: string | symbol,
+    subscriber: Subscriber,
+    context?: SubscriberContext
+  ): this
+}
+```
+
+Here's an example of how you can subscribe to an event using the `EventBusService`:
+
+```ts title=src/subscribers/my.ts
+import { EventBusService } from "@medusajs/medusa"
+
+class MySubscriber {
+ constructor({
+  eventBusService: EventBusService,
+ }) {
+  eventBusService.subscribe("product.created", (data) => {
+   // TODO handle event
+   console.log(data.id)
+  })
+ }
+}
+```
+
+In the constructor of a subscriber, you use the `EventBusService` to subscribe to the event `product.created`. In the handler method, you can perform a task every time the product is created. Notice how the handler method accepts the `data` as a parameter as explain in the previous section.
+
+:::note
+
+You can learn more about how to create a subscriber in [this documentation](../subscribers/create-subscriber.md)
+
+:::
+
+---
+
+## Processing Events
+
+In the `EventBusService` service, the `worker_` method defines the logic run for each event emitted into the queue.
+
+By default, all handler methods to that event are retrieved and, for each of the them, the stored data provided as a second parameter in `emit` is passed as an argument.
+
+---
+
+## Retrying Handlers
+
+A handler method might fail to process an event. This could happen because it communicates with a third party service currently down or due to an error in its logic.
+
+In some cases, you might want to retry those failed handlers.
+
+As briefly explained earlier, you can pass options when emitting an event as a third argument that are used to configure how the queue worker processes your job. If you pass  `attempts` upon emitting the event, the processing of a handler method is retried when it fails.
+
+Aside from `attempts`, there are other options to futher configure the retry mechanism:
+
+```ts
+type EmitOptions = {
+ delay?: number
+ attempts: number
+ backoff?: {
+  type: "fixed" | "exponential"
+  delay: number
+ }
+}
+```
+
+Here's what each of these options mean:
+
+- `delay`: delay the triggering of the handler methods by a number of milliseconds.
+- `attempts`: How many times should a subscriber handler be retried when it fails.
+- `backoff`: ...
```

Comment:
@olivermrbl I couldn't find details about this, can you just add a general meaning of what it's used for?